packages/@ourworldindata/grapher/src/facetChart/FacetChart.tsx

```diff
@@ -446,10 +455,14 @@ export class FacetChart
         return this.intermediatePlacedSeries.map((series, i) => {
             const chartInstance = intermediateChartInstances[i]
             const { xAxis, yAxis } = chartInstance
-            const { bounds: initialGridBounds, edges } = gridBoundsArr[i]
+            const {
+                bounds: initialGridBounds,
+                gridEdges,
+                cellEdges,
+            } = gridBoundsArr[i]
             let bounds = initialGridBounds
             // Only expand bounds if the facet is on the same edge as the shared axes
-            for (const edge of edges) {
+            for (const edge of gridEdges) {
                 bounds = bounds.expand({
                     [edge]: sharedAxesSizes[edge],
                 })
```

Comment:
I was puzzling over what this code is supposed to do, and now found out that we don't want to do this for the `bottom` edge!

This screenshot here is with shared y axis, meaning that the y axis ranges should be the same across all facets. And sadly, as you can see, they are not!
![CleanShot 2024-06-27 at 12 25 39](https://github.com/owid/owid-grapher/assets/2641501/f5fe4a8d-e57f-417f-817d-2bb79d686a0f)

---

For the shared y axis, we want the facets to all have the same "charting area size" (i.e. x/y axis range), but then the charts in the first column need some extra space "on the left" to account for the y axis width. That's what this code is doing.

However, for the x axis, the axis is the same height no matter whether it's hidden or not, so no need to expand the bounds here.

I think the fix is just making sure that `sharedAxisSizes["bottom"] = 0`? Not sure.

Hope that comment makes sense, it's not trivial to explain this :)

Reply:
Also, given this, it makes just as much sense to iterate over `cellEdges`, which is the one that is more correct for this use case anyway (and then we could also get rid of `gridEdges` entirely probably?).

It won't make a difference now, but again, seems like the more correct approach.

```suggestion
            for (const edge of cellEdges) {
                bounds = bounds.expand({
                    [edge]: sharedAxesSizes[edge],
                })
```

Reply:
Good catch! Setting `sharedAxisSizes.bottom` to zero does the trick, and I also removed `gridEdges` :)